### PR TITLE
[REEF-1987] Visual Studio 2017 Project Conversion

### PR DESCRIPTION
**Pull request by singlis**
The goal of these changes is to migrate over to the new DotNet projects.
Part of this effort is to be able to execute the build and tests via the
build system. Therefore with this change, the DotNet projects now build
.net452 by default and target the bridge. The build and tests can now
run using msbuild -- almost identical to the commands that are in the
appveyor.yml:
 -msbuild lang\cs\Org.Apache.REEF.DotNet.sln /t:Restore
 -msbuild lang\cs\Org.Apache.REEF.DotNet.sln
 -msbuild lang\cs\TestRunner.DotNet.proj

If there is a need to continue to use .net core builds, the
EnableDotNetBuild variable can be set to make this happen.

This completes the following sub-tasks:
   [REEF-1986] DotNet project needs to support different build projects
    - Added EnableDotNetBuild parameter in build.Common.DotNet.props.
    When active, the projects will build .net452 and .netcore but will
    target the Bridge.CLR project. If false, will target .net452 and
    existing bridge project. Defaults to false.
   [REEF-1985] Update DotNet projects to target .net 452
    - Updated all *DotNet projects to target .net 452
   [REEF-1984] Port test-runner project to the DotNet project
    - Added a DotNet test runner project, works like the old test runner
    but supports the new projects. This project can be ran using
    msbuild.
   [REEF-1983] DotNet client project supports embedding jar files as
   resource
    - Client project has changed to run the Custom code task that will
    embed the bridge exe and java jar files as resources into the file.
   [REEF-1982] DotNet port of Bridge and BridgeJAR projects
    - This add DotNet versions of the Bridge and BridgeJar projects. The
    DotNet build will now build the java project and copy jar files
    over. Bridge project builds as a .net452 exe.

JIRA:
    [REEF-1986](https://issues.apache.org/jira/browse/REEF-1986)
    [REEF-1985](https://issues.apache.org/jira/browse/REEF-1985)
    [REEF-1984](https://issues.apache.org/jira/browse/REEF-1984)
    [REEF-1983](https://issues.apache.org/jira/browse/REEF-1983)
    [REEF-1982](https://issues.apache.org/jira/browse/REEF-1982)

Pull Request:
    This closes #

### CODE REVIEW

File: lang/cs/build.Common.DotNet.props [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  </PropertyGroup>
+
+  <!-- For linux builds, always enable the dot net build -->
+  <PropertyGroup Condition="'$(OS)' != 'Windows_NT'">
+    <EnableDotNetBuild>true</EnableDotNetBuild>
+  </PropertyGroup>
+
+  <!-- For window builds, we are defaulting the dot net build to false until dot net core is fully supported -->
+  <PropertyGroup Condition="'$(OS)' == 'Windows_NT'">
+    <EnableDotNetBuild>true</EnableDotNetBuild>
```
Comment (singlis) [inline lang/cs/build.Common.DotNet.props]: Note this is supposed to be false. Will fix once I have more feedback.

File: lang/cs/Org.Apache.REEF.Bridge.CLR/Org.Apache.REEF.Bridge.CLR.DotNet.csproj [hunk 2]
```diff
@@ -36,7 +36,7 @@ under the License.
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
```
Comment (singlis) [inline lang/cs/Org.Apache.REEF.Bridge.CLR/Org.Apache.REEF.Bridge.CLR.DotNet.csproj]: Need to add build.DotNet.targets to this file,

Comment (dougmsft) [reply lang/cs/Org.Apache.REEF.Bridge.CLR/Org.Apache.REEF.Bridge.CLR.DotNet.csproj]: Please make the above changes so I can test against them.

**Comment by dougmsft** [comment]
Please make the above changes so I can test against them.

**Comment by singlis** [comment]
This is now updated.

**Pull request closed without merging**